crux-mir/lib/libcore/crucible/concurrency.rs

```diff
@@ -0,0 +1,62 @@
+//! Crucible concurrency intrinsics
+
+
+// Signal a possible preemption point. `ro` is true if this
+// preemption point MUST be read only. It is always safe to
+// pass `false`.
+#[unstable(feature = "crucible_intrinsics", issue = "none")]
+pub fn sched_yield<T>(ro: bool, x: T) {}
```

Comment:
Looking at the implementations, it seems `sched_yield`, `mutex_lock`, and `mutex_unlock` all require `x` to be represented as `MirReferenceRepr`.  Is it possible to change the signatures here to take `x: *const T` to reflect that?  I think all references and pointers are coercible to `*const T` so it hopefully won't require changes elsewhere in the code.  If it would be a pain to change then please add a comment here explaining the requirements on `T`.